scheduler/test/util.go

```diff
@@ -71,8 +74,27 @@ func (h *FakeHandle) Parallelizer() (p parallelize.Parallelizer) {
 	panic("unimplemented")
 }
 
-func (h *FakeHandle) GetWaitingPod(uid types.UID) (w framework.WaitingPod) {
-	panic("unimplemented")
+func (h *FakeHandle) GetWaitingPod(uid types.UID) framework.WaitingPod {
+	if uid == types.UID("handle-test") {
+		return nil
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: apimeta.ObjectMeta{
+			Name:      "handle-pod",
+			Namespace: "test",
+			UID:       uid,
+		},
+		Spec: v1.PodSpec{NodeName: NodeSmall.Name},
+	}
+
+	waitingPod := &waitingPod{
+		pod:            pod,
+		pendingPlugins: make(map[string]*time.Timer),
+	}
+
+	h.GetWaitingPodValue = waitingPod
+	return waitingPod
```

Comment:
Well... This function makes the test a bit hard to understand.

Can we do this instead?

```suggestion
// GetWaitingPod returns PodForHandleTest only when the uid is handle-test.
func (h *FakeHandle) GetWaitingPod(uid types.UID) framework.WaitingPod {
	if uid != types.UID("handle-test") {
		return nil
	}

	waitingPod := &waitingPod{
		pod:            PodForHandleTest,
		pendingPlugins: make(map[string]*time.Timer),
	}

	h.GetWaitingPodValue = waitingPod
	return waitingPod
```

And, at the test case, you have to swap PodSmall and PodForHandleTest.
Then, you should be able to remove `PodForHandlePod` too. (can use `PodForHandleTest` at `makeTestWaitingPod` instead)

Reply:
I appreciate your feedback. That's simple and clean.